src/realtime/pubsub/events.rs

```diff
@@ -123,33 +123,41 @@ pub async fn events_ws(
     let socket_id = nanoid::nanoid!();
 
     {
-        let pubsub = pubsub.clone();
         let socket_id = socket_id.clone();
-
+        let pubsub = pubsub.clone();
         on_message_handler = Box::new(move |msg: String| {
-            if let Ok(esr) = serde_json::from_str::<EventSocketRequest>(&msg) {
-                use EventSocketRequest::*;
-                match esr {
-                    Subscribe { topics } => {
-                        for topic in topics {
-                            pubsub.subscribe(&socket_id, topic).unwrap_or(());
+            let socket_id = socket_id.clone();
+            let pubsub = pubsub.clone();
```

Comment:
!!!

Reply:
This is perfectly fine and is the expected pattern. The call to
`Data<PubSub>::clone()` is also fine because `actix_web::Data<T>` is internally
equivalent to `Arc<T>`, so the clone is cheap.

I also don't like that you have to redefine variables right outside of a `move`
closure but this is the accepted way of doing it I guess.

LGTM